app/Api/Auth/ApiAuthorizer.php

```diff
@@ -48,6 +48,8 @@ protected function checkRequestMethod(): bool
      */
     protected function checkApiKey(): bool
     {
-        return hash_equals($this->api_key, $_GET['api_key']);
+        $user_api_key = $_GET['api_key'];
+        \assert(\is_string($user_api_key));
```

Comment:
Aren't our asserts in this case just used as indicators for static analysers (IDEs and such) what to expect from the code at that point? As the assert functionality is probably disabled anyways in the production version / when hosting. 

At least if it's anything like in Java.

Reply:
Yeah, I also read that `assert` is mostly disabled in production.
But if `assert` is basically only an indicator for tools like phan, this does not really fix the potential errors we might see.
Therefore, wouldn't it be better to make sure types are 100% correct by providing such fallbacks as mentioned above?

Reply:
1. some functions are implemented both as function and as optimized opcode in the php-engine.
such functions must be fully-qualified (using a \ at the start) to be recognized as such a "special" function by the php-compiler during the compilation phase. thats why php-cs-fixer adds a leading \ to some functions to enable this optimization.

2. assert is used to tell phan that a variable has a specific type. in this case, phan complains that $_GET['api_key'] could be of type "string" or "array" (because `api_key` could be passed as `api_key[]=foo`)

Reply:
Oh, ok. Then everything is fine. 😃 
Thank you for the explanation.

Reply:
take a look at this opcodes here:
https://3v4l.org/dD6Er/vld#output

the first strlen() is called as foo\strlen (line 4), the second call is completely optimized away (line 9) :)